src/estimagic/inference/bootstrap.py

```diff
@@ -17,9 +19,10 @@
 
 
 def bootstrap(
-    data,
     outcome,
     *,
+    data=None,
+    existing_outcomes=None,
     outcome_kwargs=None,
     n_draws=1_000,
```

Comment:
Currently the total number of draws is `n_draws + len(existing_outcomes)`, but it is not clearly documented in the docstring.

1. We should document the precise meaning of `n_draws` in the docstring
2. I think it would be if `n_draws` is the number of draws used. If existing outcomes is longer than `n_draws` only the first part of `existing_outcomes` should be used. Otherwise, `n_draws - len(existing_outcomes)` new draws should be sampled.